site/learn/expert-advice/authentication/webauthn-explained.md

```diff
@@ -0,0 +1,515 @@
+
+### Passwords
+
+Passwords are a solid choice, but can be forgotten, stolen or re-used. In fact, over 11B username/password pairs are publicly available on the site HaveIBeenPwned. User's aren't really fans of remembering passwords either, and insecurely re-use them across different systems.
+
+This means that user accounts on your system don't simply depend on your security posture, but on the posture of all the other sites that the user has registered for with the same password.
+
+Doh!
+
+### Password Managers
+
+Well, what about password managers?
+
+The user experience can be tough when dealing with password managers. If you're using a password manager that integrates into the browser, any non-standard login form UX can be difficult for it to handle.
+
+If a password manager is centralized, it is now a high-value target. If it is not, then users can't login to accounts from more than one device.
+
+### MFA
+
+If you add an additional factor of authentication (i.e. text the user a code, send them an email with a code, or require them to enter some code based on some algorithm using something like Google Authenticator), you can increase security. These additional factors of the authentication add additional assurance about who the user is during an authentication process.
+
+However, the user experience is suboptimal. It can be a hassle to force someone to go to a different device to get a code.
+
+Depending on the method, it can also be insecure. For example, SMS codes can be phished. If you are using push messaging, attackers can bombard users until [MFA fatigue sets in](https://techcommunity.microsoft.com/t5/microsoft-entra-azure-ad-blog/defend-your-users-from-mfa-fatigue-attacks/ba-p/2365677).
+
+### Single Sign-on
+
+Single sign-on is another option, which doesn't have these same risks. To be more precise, it centralizes them. This makes sense, as most people pay more attention to their Google/Facebook/Apple account than to the many random accounts they've created on other sites. You can depend on a big company's engineers to encourage best practices.
+
+However, single sign-on has a business risk. You are now depending on a third party to allow your users to access your website.
+
+## WebAuthn
+
+WebAuthn is a decentralized, consumer-friendly, secure alternative for protecting your clownwear store.
+
+With WebAuthn, the private keys which authenticate the user to the website typically remain on the authenticator. (There are some services that sync them across devices, but that is not part of the specification.) The server stores only public keys, which means that, unlike password hashes, there's not much value in obtaining them.
```

Comment:
The [Level 3 version](https://w3c.github.io/webauthn/) of the spec (still in draft status) has more info about [credential backup](https://w3c.github.io/webauthn/#sctn-credential-backup), and adds flags to the authenticator response to indicate whether a credential is eligible for backup, and whether it has been backed up.

Reply:
The spec does not dictate the methods for sharing passkeys, but it has expanded metadata to support the use case.

Reply:
Good to know, but I think that might be in the weeds a bit for this.